### ISSUE TITLE
navigate to newer version of the docs when you click search result in old version

### ISSUE THREAD

**Issue by Sboonny** [gatsbyjs/gatsby#38686]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Summary

When I tried to search for routing in old versions of the docs, I always navigated to version 5, although the info I need are related to version 3.

https://github.com/gatsbyjs/gatsby/assets/88248797/17bd7f9f-8688-44f4-bc85-7f23ee740fc7

I had to use the sidebar and find the needed info, to make sure that the info still relevant.

### Steps to Resolve this Issue

I have no idea how to solve this 😞.

But to resolve it, when someone navigate using search in docs it should:

1. check docs version.
2. navigate to related info about that version.

**Comment by alokVerma749** [comment]
https://v3.gatsbyjs.com/docs/  
this way particular version docs can be accessed
[v3 doc link](https://v3.gatsbyjs.com/docs/)

**Comment by alokVerma749** [comment]
> ### Preliminary Checks
> * [x]  This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
> * [x]  This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions
> 
> ### Summary
> When I tried to search for routing in old versions of the docs, I always navigated to version 5, although the info I need are related to version 3.
> 
>  documentation-routing.mp4 
> I had to use the sidebar and find the needed info, to make sure that the info still relevant.
> 
> ### Steps to Resolve this Issue
> I have no idea how to solve this 😞.
> 
> But to resolve it, when someone navigate using search in docs it should:
> 
> 1. check docs version.
> 2. navigate to related info about that version.

yes you are correct, we have to take the desired version and put it in query string / URL

**Comment by jaivardhan-bhola** [comment]
Can I work on this

**Comment by alokVerma749** [comment]
how would you approach this issue??

**Comment by jaivardhan-bhola** [comment]
have a separate list for old version docs and add them to search and then filter via versions
